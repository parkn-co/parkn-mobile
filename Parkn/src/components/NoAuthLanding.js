// @flow

import React from 'react';
import {View, TouchableOpacity, Text, StyleSheet} from 'react-native';
import {bluePalette, grayPalette} from 'styles/colors';
import {fontNames} from 'styles/fonts';

const NoAuthLanding = ({go}: {go: Function}): React.Element<*> => (
  <View style={styles.container}>
    <View style={styles.sectionWrapper}>
      <Text style={styles.logo}>{'Parkn'}</Text>
    </View>
    <View style={styles.sectionWrapper}>
      <TouchableOpacity style={styles.button} onPress={() => go.to({id: 'SignUp'})}>
        <Text style={styles.buttonText}>{'Sign Up'}</Text>
      </TouchableOpacity>
      <TouchableOpacity style={styles.button} onPress={() => go.to({id: 'SignIn'})}>
        <Text style={styles.buttonText}>{'Sign In'}</Text>
      </TouchableOpacity>
    </View>
  </View>
);

export default NoAuthLanding;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: bluePalette.medium,
  },
  sectionWrapper: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 15,
  },
  logo: {
    color: grayPalette.white,
    opacity: 0.75,
    fontFamily: fontNames.moonBold,
    fontSize: 50,
  },
  button: {
    paddingTop: 20,
    paddingBottom: 20,
    borderWidth: 1,
    borderColor: 'rgba(255,255,255,0.5)',
    alignSelf: 'stretch',
    alignItems: 'center',
    justifyContent: 'center',
    marginBottom: 15,
  },
  buttonText: {
    color: grayPalette.white,
    opacity: 0.75,
    fontFamily: fontNames.light,
    fontSize: 20,
  },
});
