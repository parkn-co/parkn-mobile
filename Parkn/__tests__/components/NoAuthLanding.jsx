/* eslint-env jest */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import NoAuthLanding from '../../src/components/NoAuthLanding';

describe('NoAuthLanding', () => {
  it('renders correctly using snapshots', () => {
    expect(renderer.create(
      <NoAuthLanding go={jest.fn()} />
    )).toMatchSnapshot();
  });
});
