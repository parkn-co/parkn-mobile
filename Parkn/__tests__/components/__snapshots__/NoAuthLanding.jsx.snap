exports[`NoAuthLanding renders correctly using snapshots 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#076FEB",
      "flex": 1,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 15,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Moon-Bold",
          "fontSize": 50,
          "opacity": 0.75,
        }
      }>
      Parkn
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 15,
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "borderColor": "rgba(255,255,255,0.5)",
          "borderWidth": 1,
          "justifyContent": "center",
          "marginBottom": 15,
          "opacity": 1,
          "paddingBottom": 20,
          "paddingTop": 20,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "OpenSans-Light",
            "fontSize": 20,
            "opacity": 0.75,
          }
        }>
        Sign Up
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "borderColor": "rgba(255,255,255,0.5)",
          "borderWidth": 1,
          "justifyContent": "center",
          "marginBottom": 15,
          "opacity": 1,
          "paddingBottom": 20,
          "paddingTop": 20,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "OpenSans-Light",
            "fontSize": 20,
            "opacity": 0.75,
          }
        }>
        Sign In
      </Text>
    </View>
  </View>
</View>
`;
